sherpa/include/sherpa/astro/xspec_extension.hh

```diff
@@ -62,6 +62,21 @@ namespace sherpa { namespace astro { namespace xspec {
 typedef sherpa::Array< float, NPY_FLOAT > FloatArray;
 typedef float FloatArrayType;
 
+// Try and support the use of std::transform while still building
+// against C++-98 compilers.
+//
+#if __cplusplus > 199711L
+#define CONVERTARRAY(orig, out, npts)					\
+        std::transform(std::begin(orig), std::end(orig), std::begin(out), \
+                       [](const double val) -> FloatArrayType { return static_cast<FloatArrayType>(val); });
+#else
+#define CONVERTARRAY(orig, out, npts)					\
+	for (int i = 0; i < npts; i++) { \
```

Comment:
when possible, you should avoid using single character variable names-  int i,   float f, etc...  Using indexes such as   int ii,   float ff  improves readability and searching (ii will generate fewer false hits than i).

Also,  neglecting optimization and loop unrolling,  reversing indexing for simple independent assignments

     for (int ii = npts; ii--; ) { }

will generally provide more efficient code than 

    for (int ii = 0; ii < npts; ii++) { } 

when compiled the same way.

Reply:
@wmclaugh - all I did here was resurrect the old code (the for loop) that we had before #859 was merged. I didn't want to spend time complicating things with changes to code that we knew already worked (and that we can hopefully remove soon). If someone wants to fix up this PR to make these changes then they are more than welcome to, but I feel I've done enough with this patch as is.